libs/ai-endpoints/tests/unit_tests/test_api_key.py

```diff
@@ -43,7 +43,7 @@ def test_create_without_api_key(public_class: type) -> None:
 
 
 def test_create_unknown_url_no_api_key(public_class: type) -> None:
-    with no_env_var("NVIDIA_API_KEY"):
+    with no_env_var("NVIDIA_API_KEY") and pytest.warns(UserWarning):
```

Comment:
for future: there could be multiple UserWarnings raised, add checks to make sure it's the warning you expect